language-plutus-napkin/src/Language/PlutusNapkin/Type.hs

```diff
@@ -43,18 +51,44 @@ data Builtin = AddInteger
              | TxHash
              | BlockNum
              | BlockTime
+             deriving (Show, Generic, NFData)
+
+data Keyword = KwIsa
+             | KwAbs
+             | KwInst
+             | KwLam
+             | KwFix
+             | KwBuiltin
+             | KwFun
+             | KwForall
+             | KwByteString
+             | KwInteger
+             | KwFloat
+             | KwSize
+             | KwType
+             deriving (Show, Generic, NFData)
+
+data Special = OpenParen
+             | CloseParen
+             | OpenBracket
+             | CloseBracket
+             deriving (Show, Generic, NFData)
 
 -- | Annotated type for names
-data Name a = LexVar a BSL.ByteString
-            | LexTyVar a BSL.ByteString
-            | LexTermVar a BSL.ByteString
-            | LexInt a Integer
-            | LexFloat a Float -- TODO check for silent truncation in the lexer
-            | LexExp a Integer
-            | LexBS a BSL.ByteString
-            | LexBuiltin a Builtin
-            | LexSize a Natural
-            | LexSizeTerm a BSL.ByteString
+data Token a = LexName { loc :: a, identifier :: Int }
+             | LexInt { loc :: a, int :: Integer }
+             | LexFloat { loc :: a, float :: Float } -- TODO check for silent truncation in the lexer
```

Comment:
As I wrote in the email, let's drop floats for now. (We decided to not include them in the first version of PN.)